### DISCUSSION TITLE
Refill parser from previous R command

### DISCUSSION THREAD

**Discussion by volloholic** [IDEMSInternational/R-Instat#9788] [Ideas]
The read-write system should enable a user to click on a previous piece of code in the output window and repopulate the dialog used to generate it.

**Comment by rdstern** [comment]
@volloholic claims this is consistent with the coding that @lloyddewit is starting.  Also that it is not so difficult if we start in a way that is a bit "cavalier".  And we improve the system in stages.

I give my understanding and assume @volloholic will edit as necessary.
a) It starts with a right-click option - perhaps called **_Dialogue_** on each of the windows that show commands, so the log, script and output windows.
b) You have first (usually) to select a command - at least.  Initially it might just be the command.
c) Then it opens a dialogue.  This has the usual data selector, i.e. with the data frame and a multiple receiver.  However, the data frame has an additional option which is blank or __none__ as there could be a dialogue that doesn't start with a data frame.
d) It also has a control with the label **_Dialogue_**.  
e) It also has the command.  This has the label **_R Command_**.  This may extend over multiple lines.  This has the command if you selected it.  It is blank if you didn't select a command.  Ideally, when in that field, you could return to the window and still select the command. (You can't usually do this.  
f) Either now, or in the future, you could select the initial lines before the command.  When they exist, the initial Comment line often includes the name of the dialogue.  In that case it might be included automatically.  The next line(s) will usually give the data frame and possibly the variables.  If so, then these would be detected and included automatically.
g) What could be really tricky is to check that this dialogue will work when you press ok.  We suggest this aspect be ignored initially.  Then, if the system works when no mistake is made, then we add checks later. 
h) I assume the new dialogue will also be put somewhere.  My initial suggestion is the tools menu.
